shared/src/TwinCollection.cs

```diff
@@ -176,9 +176,11 @@ public Metadata GetMetadata()
         /// Gets the LastUpdated time for this property
         /// </summary>
         /// <returns>DateTime instance representing the LastUpdated time for this property</returns>
+        /// <exception cref="System.ArgumentNullException">Thrown when the TwinCollection metadata is null. 
```

Comment:
I don't see this exception being thrown anywhere in this method.

Reply:
IOW, should we remove the doc comment that says this could return this exception?

Reply:
This exception gets thrown if metadata is null and someone calls this method so the comment is still valid.

For example a customer calls default TwinCollection constructor with no parameters, metadata would be null and then calling this method would return this exception

Reply:
The code as I see it returns null, not an exception.

Reply:
I see, it happens because casting a null to DateTime throws that exception.

Changing the exception thrown is also a breaking change. We can't do that.